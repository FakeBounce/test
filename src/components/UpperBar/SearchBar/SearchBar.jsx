import React, { Component } from "react";
import { translate } from "common/methods/translations";

import styles from "./SearchBar.module.scss";
import { Collapse, Input } from "reactstrap";
import { observable } from "mobx";
import { observer } from "mobx-react";

@observer
class SearchBar extends Component {
  @observable searchInput = "";

  constructor() {
    super();
    this.handleSearchChange = this.handleSearchChange.bind(this);
  }

  handleSearchChange(event) {
    if (event && event.target && event.target.value)
      this.searchInput = event.target.value;
  }

  onKeyPressEvent(e) {
    const { history } = this.props;
    const key = e.which || e.keyCode;
    if (key === 13 && this.searchInput !== "") {
      history.push(`/globalSearch?phrase=${this.searchInput}`);
    }
  }

  render() {
    const { isOpen } = this.props;
    return (
      <div className={styles.GlobalSearch}>
        <Collapse isOpen={isOpen}>
          <Input
            placeholder={translate("globals.header.search")}
            value={this.searchInput}
            onChange={this.handleSearchChange}
            onKeyPress={this.onKeyPressEvent.bind(this)}
            ref={input => input && input.focus()}
          />
        </Collapse>
      </div>
    );
  }
}

export default SearchBar;
