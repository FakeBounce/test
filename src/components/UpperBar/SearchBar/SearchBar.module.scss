@import '~common/styles/variables';

.GlobalSearch {
  position: relative;
  z-index: -1;
}

.GlobalSearch div {
  &:global(.collapse) {
    position: absolute;
    width: calc(100% - 60px);
    right: 0;
    margin-left: 60px;
    background-color: $white;
    box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.24);
  }
  &:global(.collapsing) {
    position: absolute;
    margin-left: 60px;
    right: 0;
    left: 0;
    background-color: $white;
    box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.24);
  }
}

.GlobalSearch input {
  width: 100%;
  font-size: 18px;
  letter-spacing: 0.09px;
  line-height: 22px;
  font-weight: 300;
  color: $light-gray;
  height: initial;
  padding: 19px 25px;
  border-radius: 0;
  border: 0;
  border-bottom: 0.5px solid $light-gray;
  margin-bottom: 50px;
  &:focus {
    border-color: $light-gray;
    box-shadow: none;
  }
}