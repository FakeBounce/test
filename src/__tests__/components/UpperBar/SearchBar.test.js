import React from "react";
import { shallow, configure } from "enzyme/build";
import Adapter from "enzyme-adapter-react-16/build";
import SearchBar from "components/UpperBar/SearchBar/SearchBar";

configure({ adapter: new Adapter() });

describe("Testing SearchBar component", () => {
  it("renders as expected without params", () => {
    const wrapper = shallow(<SearchBar />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders as expected with params", () => {
    const history = [];
    history.push("/home");
    const wrapper = shallow(<SearchBar history={history} isOpen />);
    expect(wrapper).toMatchSnapshot();
  });

  it("functions behave correctly", () => {
    const history = [];
    history.push("/home");
    history.push("/home/test");
    history.push("/home");
    const wrapper = shallow(<SearchBar history={history} />);
    expect(history.length).toBe(3);
    wrapper.instance().handleSearchChange({ target: { value: "yes" } });
    wrapper.instance().onKeyPressEvent({ keyCode: 13 });
    expect(history.length).toBe(4);
    expect(history[3]).toBe("/globalSearch?phrase=yes");
  });
});
