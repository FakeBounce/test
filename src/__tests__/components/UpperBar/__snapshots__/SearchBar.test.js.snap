// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SearchBar component renders as expected with params 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    history={
        Array [
            "/home",
          ]
    }
    isOpen={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Collapse
        appear={false}
        enter={true}
        exit={true}
        in={false}
        isOpen={true}
        mountOnEnter={false}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        tag="div"
        timeout={350}
        unmountOnExit={false}
>
        <Input
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder="Rechercher"
                type="text"
                value=""
        />
</Collapse>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "appear": false,
        "children": <Input
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder="Rechercher"
          type="text"
          value=""
/>,
        "enter": true,
        "exit": true,
        "in": false,
        "isOpen": true,
        "mountOnEnter": false,
        "onEnter": [Function],
        "onEntered": [Function],
        "onEntering": [Function],
        "onExit": [Function],
        "onExited": [Function],
        "onExiting": [Function],
        "tag": "div",
        "timeout": 350,
        "unmountOnExit": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onChange": [Function],
          "onKeyPress": [Function],
          "placeholder": "Rechercher",
          "type": "text",
          "value": "",
        },
        "ref": [Function],
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Collapse
          appear={false}
          enter={true}
          exit={true}
          in={false}
          isOpen={true}
          mountOnEnter={false}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          tag="div"
          timeout={350}
          unmountOnExit={false}
>
          <Input
                    onChange={[Function]}
                    onKeyPress={[Function]}
                    placeholder="Rechercher"
                    type="text"
                    value=""
          />
</Collapse>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "appear": false,
          "children": <Input
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Rechercher"
            type="text"
            value=""
/>,
          "enter": true,
          "exit": true,
          "in": false,
          "isOpen": true,
          "mountOnEnter": false,
          "onEnter": [Function],
          "onEntered": [Function],
          "onEntering": [Function],
          "onExit": [Function],
          "onExited": [Function],
          "onExiting": [Function],
          "tag": "div",
          "timeout": 350,
          "unmountOnExit": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onChange": [Function],
            "onKeyPress": [Function],
            "placeholder": "Rechercher",
            "type": "text",
            "value": "",
          },
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Testing SearchBar component renders as expected without params 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Collapse
        appear={false}
        enter={true}
        exit={true}
        in={false}
        isOpen={false}
        mountOnEnter={false}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        tag="div"
        timeout={350}
        unmountOnExit={false}
>
        <Input
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder="Rechercher"
                type="text"
                value=""
        />
</Collapse>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "appear": false,
        "children": <Input
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder="Rechercher"
          type="text"
          value=""
/>,
        "enter": true,
        "exit": true,
        "in": false,
        "isOpen": false,
        "mountOnEnter": false,
        "onEnter": [Function],
        "onEntered": [Function],
        "onEntering": [Function],
        "onExit": [Function],
        "onExited": [Function],
        "onExiting": [Function],
        "tag": "div",
        "timeout": 350,
        "unmountOnExit": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onChange": [Function],
          "onKeyPress": [Function],
          "placeholder": "Rechercher",
          "type": "text",
          "value": "",
        },
        "ref": [Function],
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Collapse
          appear={false}
          enter={true}
          exit={true}
          in={false}
          isOpen={false}
          mountOnEnter={false}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          tag="div"
          timeout={350}
          unmountOnExit={false}
>
          <Input
                    onChange={[Function]}
                    onKeyPress={[Function]}
                    placeholder="Rechercher"
                    type="text"
                    value=""
          />
</Collapse>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "appear": false,
          "children": <Input
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Rechercher"
            type="text"
            value=""
/>,
          "enter": true,
          "exit": true,
          "in": false,
          "isOpen": false,
          "mountOnEnter": false,
          "onEnter": [Function],
          "onEntered": [Function],
          "onEntering": [Function],
          "onExit": [Function],
          "onExited": [Function],
          "onExiting": [Function],
          "tag": "div",
          "timeout": 350,
          "unmountOnExit": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onChange": [Function],
            "onKeyPress": [Function],
            "placeholder": "Rechercher",
            "type": "text",
            "value": "",
          },
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;
